rpc/grpc/client_server.go

```diff
@@ -92,16 +97,29 @@ func StartBlockAPIGRPCClient(protoAddr string, opts ...grpc.DialOption) (BlockAP
 // StartBlobstreamAPIGRPCClient dials the gRPC server using protoAddr and returns a new
 // BlobstreamAPIClient.
 func StartBlobstreamAPIGRPCClient(protoAddr string, opts ...grpc.DialOption) (BlobstreamAPIClient, error) {
+	parsedAddr := ParseProtoAddr(protoAddr)
+
 	if len(opts) == 0 {
 		opts = append(opts, grpc.WithTransportCredentials(insecure.NewCredentials()))
 	}
 	opts = append(opts, grpc.WithContextDialer(dialerFunc))
-	conn, err := grpc.Dial( //nolint:staticcheck
-		protoAddr,
+	conn, err := grpc.NewClient(
+		parsedAddr,
 		opts...,
 	)
 	if err != nil {
 		return nil, err
 	}
 	return NewBlobstreamAPIClient(conn), nil
 }
+
+// ParseProtoAddr parses the protoAddr and returns the address without the prefix
+func ParseProtoAddr(protoAddr string) string {
+	// Single regex to handle all schemes:
+	// - dns:host:port -> host:port
+	// - unix:///path -> /path (removes unix: and up to 2 extra slashes)
+	// - tcp://host:port -> host:port
+	// - any://something -> something
+	re := regexp.MustCompile(`^(?:dns:|unix:/{0,2}|[a-zA-Z][a-zA-Z0-9+.-]*://)`)
+	return re.ReplaceAllString(protoAddr, "")
```

Comment:
actually, `unix://` and `dns://` are supported by the gRPC client, they shouldn't be stripped. However, the rest should be. Please update the implementation along with the comment and move the comment to be a go doc